app/controllers/common/RemoveEventController.scala

```diff
@@ -57,9 +54,9 @@ class RemoveEventController @Inject()(
         value => {
           val originalUserAnswers = request.userAnswers
           if (value) {
-            val removedUserAnswers = originalUserAnswers.removeWithPath(JsPath \ s"event${eventType.toString}")
-              //TODO: Will be replaced with new BE endpoint created in PODS-8597
-                Future.successful(Redirect(RemoveEventPage(eventType).navigate(waypoints, originalUserAnswers, removedUserAnswers).route))
+            eventReportingConnector.compileEvent(request.pstr, request.userAnswers.eventDataIdentifier(eventType), currentVersion = request.userAnswers.get(VersionInfoPage).map(_.version).getOrElse(0), delete = true).flatMap { _ =>
```

Comment:
Could we use the CompileService here instead of going direct to the connector? The idea is all compiles should go thru compile service.